examples/fabric-admin/rpc/RpcClient.cpp

```diff
@@ -86,3 +100,27 @@ CHIP_ERROR AddSynchronizedDevice(chip::NodeId nodeId)
 
     return CHIP_NO_ERROR;
 }
+
+CHIP_ERROR RemoveSynchronizedDevice(chip::NodeId nodeId)
+{
+    ChipLogProgress(NotSpecified, "RemoveSynchronizedDevice");
+
+    if (removeSynchronizedDeviceCall.active())
+    {
+        ChipLogError(NotSpecified, "Remove Synchronized Device operation is in progress\n");
+        return CHIP_ERROR_BUSY;
+    }
+
+    chip_rpc_SynchronizedDevice device;
+    device.node_id = nodeId;
+
+    // The RPC will remain active as long as `removeSynchronizedDeviceCall` is alive.
```

Comment:
I do not understand this comment. Could you explain why the active check below is correct? This somehow feels like a race condition however not really sure and I would hope that comments would explain this.

The way I read the code is that we issue an RPC (RemoveSynchronizedDevice) and then we validate that RPC is still on-going immediately after issuing it. Why is that correct? What if we get a context switch and the RPC is really really fast? Is the RPC guaranteed to not be finished (e.g. it also runs under the chip stack so will not progress while other code is running)?

Reply:
Updated the comments:

    // By assigning the returned call to the global 'removeSynchronizedDeviceCall', the RPC
    // call is kept alive until it completes. When a response is received, it
    // will be logged by the handler function and the call will complete.
    removeSynchronizedDeviceCall = fabricBridgeClient.RemoveSynchronizedDevice(device, OnRemoveDeviceResponseCompleted);

    if (!removeSynchronizedDeviceCall.active())
    {
        // The RPC call was not sent. This could occur due to, for example, an invalid channel ID. Handle if necessary.
        return CHIP_ERROR_INTERNAL;
    }